generators/client/templates/angular/src/main/webapp/app/shared/login/login.component.ts.ejs

```diff
@@ -71,8 +71,8 @@ export class <%=jhiPrefixCapitalized%>LoginModalComponent implements AfterViewIn
         }).then(() => {
             this.authenticationError = false;
             this.activeModal.dismiss('login success');
-            if (this.router.url === '/register' || (/^\/activate\//.test(this.router.url)) ||
-                (/^\/reset\//.test(this.router.url))) {
+            if (this.router.url === '/register' || this.router.url.startsWith('/activate/') ||
```

Comment:
why did you change it?

Reply:
It reports an error due to typescript eslint [rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-string-starts-ends-with.md) that's enabled by default as recommended configuration. I also prefer `startsWith` over regex in such scenarios as it's more readable.